zerver/lib/subscription_info.py

```diff
@@ -57,6 +57,7 @@ def get_next_color() -> str:
 
         # Add versions of the Subscription fields based on a simulated
         # new user subscription set.
+        archived = False
```

Comment:
This seems wrong -- `archived` is not a personal user setting -- it should just read this one out of `stream.deactivated`.

Reply:
Yes you are right about that. That line should come under stream fields and not user subscription set.
I directly assigned `false` instead of getting value from `stream.deactivated` as web-public streams will never be `archived` but yeah maybe hardcoding here is not the right thing to do.